spec.bs

```diff
@@ -1750,26 +1747,26 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
               |perBuyerTimeout| to |perBuyerCumulativeTimeout|.
             1. Let |generateBidStartTime| be |settings|'s
               [=environment settings object/current monotonic time=].
-            1. Set « |generatedBid|, |bidDebugReportInfo| » to the result of [=generate a bid=] given
+            1. Set « |generatedBids|, |bidDebugReportInfo| » to the result of [=generate potentially multiple bids=] given
               |allTrustedBiddingSignals|, |auctionSignals|, a [=map/clone=] of |browserSignals|,
-              |perBuyerSignals|, |directFromSellerSignalsForBuyer|, |perBuyerTimeout|,
-              |expectedCurrency|, |ig|, and |auctionStartTime|.
+              |perBuyerSignals|, |directFromSellerSignalsForBuyer|, |perBuyerTimeout|, |expectedCurrency|,
+              1 (for multiBidLimit), |ig|, and |auctionStartTime|.
             1. Set |ig|'s [=interest group/ads=] to |originalAds|.
             1. Set |ig|'s [=interest group/ad components=] to |originalAdComponents|.
             1. Let |generateBidDuration| be the [=duration from=] |generateBidStartTime| to
               |settings|'s [=environment settings object/current monotonic time=], in milliseconds.
             1. If |perBuyerCumulativeTimeout| is not null, then decrement |perBuyerCumulativeTimeout|
               by |generateBidDuration|.
-            1. If |generatedBid| is a [=generated bid=]:
+            1. [=Assert=] that [=list/size=] of |generatedBids| &le; 1
+            1. [=list/For each=] |generatedBid| of |bidsBatch|:
               1. [=Assert=] that [=query generated bid k-anonymity count=] given |generatedBid| returns true.
+              1. [=Apply any component ads target to a bid=] given |generatedBid|.
               1. [=list/Append=] |generatedBid| to |bidsToScore|.
-              1. [=Register a bid for forDebuggingOnly reports=] given |generatedBid|, |bidDebugReportInfo| and
-                 |bidDebugReportInfoList|.
-            1. If |generatedBid| is null, [=register a bid for forDebuggingOnly reports=] given null,
-                 |bidDebugReportInfo| and |bidDebugReportInfoList|.
+          1. [=Register bids for forDebuggingOnly reports=] given |bidsToScore|, |bidDebugReportInfo|, and
+               |bidDebugReportInfoList|
           1. [=list/For each=] |bidToScore| of |bidsToScore|:
             1. If |bidToScore|'s [=generated bid/for k-anon auction=] is true,
-                [=list/append=] |bidToScore|'s [=generated bid/interest group=] to |bidIgs|.
+                [=list/insert=] |bidToScore|'s [=generated bid/interest group=] in |bidIgs|.

```

Comment:
No idea why I did it, and looks incorrect since insert wants a position? Though maybe bidIgs should be a set?
Well, that seems like a totally different change.